frontend/packages/dev-console/src/components/hpa/hpa-utils.ts

```diff
@@ -159,3 +160,22 @@ export const getInvalidUsageError = (
 
   return null;
 };
+
+export const doesHpaMatch = (workload: K8sResourceCommon) => (
+  thisHPA: HorizontalPodAutoscalerKind,
+) => {
+  const workloadAPI: string = workload.apiVersion;
+  const workloadKind: string = workload.kind;
+  const workloadName: string = workload.metadata.name;
+  const ref = thisHPA.spec.scaleTargetRef;
```

Comment:
will spec always be there? based on type `HorizontalPodAutoscalerKind` it looks that way but you did check here https://github.com/openshift/console/pull/6150/files#diff-c58b3491cc7ee3a8e71262e1ccf25712R175

Reply:
Hmm... that is an optional param. I can check against a missing `spec` I guess. At a certain point the API of a function should be respected - guess that point is somewhat arbitrary.